suite-native/module-connect-device/src/components/PinForm.tsx

```diff
@@ -14,17 +16,32 @@ const pinMatrix = [
     [1, 2, 3],
 ];
 
+const MATRIX_MARGIN = 40;
+
 const cardStyle = prepareNativeStyle(utils => ({
     marginBottom: utils.spacings.small,
-    padding: 40,
+    padding: MATRIX_MARGIN,
 }));
 
 const pinProgressWrapperStyle = prepareNativeStyle(utils => ({
     height: utils.spacings.extraLarge,
     justifyContent: 'center',
 }));
 
+const loaderWrapperStyle = prepareNativeStyle(utils => ({
+    position: 'absolute',
+    width: '100%',
+    height: '100%',
+    margin: MATRIX_MARGIN,
+    paddingBottom: utils.spacings.extraLarge,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+}));
+
 export const PinForm = () => {
+    const [isPinFormSubmitting, setIsPinFormSubmitting] = useState(false);
```

Comment:
You could use `isSubmitting` from https://react-hook-form.com/docs/useform/formstate.

Reply:
Then you can avoid props passing and get this value from the form context

Reply:
I can't, I need custom logic because pin verification is not a promise. I think I've named it wrong, it's actually not "form submitting" but "pin verification processing"

Reply:
Oh yeah, forgot about that.

Reply:
Thanks for better idea ❤️  I shouldn't have used isSubmitting, but isSubmitted works fine and looks much better 👍 
185edffb997908bb769c29506903e82daa976365